core/src/main/java/org/elasticsearch/action/ingest/IngestActionFilter.java

```diff
@@ -91,8 +92,8 @@ void processIndexRequest(Task task, String action, ActionListener listener, Acti
             chain.proceed(task, action, indexRequest, listener);
             return;
         }
-        executionService.execute(indexRequest, pipelineId, t -> {
-            logger.error("failed to execute pipeline [{}]", t, pipelineId);
+        executionService.execute(indexRequest, t -> {
+            logger.error("failed to execute pipeline [{}]", t, indexRequest.pipeline());
             listener.onFailure(t);
         }, success -> {
             indexRequest.putHeader(PIPELINE_ALREADY_PROCESSED, true);
```

Comment:
+1 this makes sense. Then we can drop the `PIPELINE_ALREADY_PROCESSED` transport header.